app.ts

```diff
@@ -0,0 +1,293 @@
+      }
+    });
+
+    tableBody = <HTMLTableSectionElement> document.getElementById("dataTableBody");
+
+    const dataRow = tableBody.insertRow(0);
+
+    //make the ajax call to retrieve the records and build the table header
+    $.ajax({
+      url: `http://localhost:2050/columns`,
+      dataType:'json',
+      async:false,
+      success:function(data) {
+  
+        //const dataRow = tableBody.insertRow(0);
+        $.each(data, function(cell)
+          {
+            const newCell = dataRow.insertCell(-1);
+            newCell.outerHTML = `<th>${data[cell]}</th>`;
+             
+          }
+        )
+      }
+    });
+
+    //call the button property set fucntion to set disable/enably buttons appropriately 
+    buttonPropertySet();
+  
+  }
+
+}
+
+class utility
+{
+  timeout:number = 0;
+
+  debounce(func: Function, wait:number){
+    //let timeout:number;
+  
+    return (...args:any[]) => {
+      const later = () => {
+        clearTimeout(this.timeout);
+        func(...args);
+      };
+  
+      clearTimeout(this.timeout);
+      this.timeout = setTimeout(later, wait);
+    };
+  };
+}
+
+const myTable = new table();
+const myUtility = new utility();
+
+window.onload = function()
+{
+
+  const request = myUtility.debounce(myTable.buildTable,250);
+
+  request(myTable.from,myTable.to);
+
+}
+
+//on resize funtionalty to rebuild the table
+$(window).on('resize', function() 
+{
+  // wrap the logic within a debounce funtion to prevent unnecesary calls. 
+  let request = myUtility.debounce(function(){
+
+    myTable.from = myTable.from;
+
+    myTable.buildTable(myTable.from,myTable.to);
+
+  },250);
+
+  request();
+}
+);
+
+function buttonPropertySet()
+{
+  let previous = <HTMLInputElement> document.getElementById("previous");
+  let previous5 = <HTMLInputElement> document.getElementById("previous5");
+  let previous10 = <HTMLInputElement> document.getElementById("previous10");
+  let next = <HTMLInputElement> document.getElementById("next");
+  let next5 = <HTMLInputElement> document.getElementById("next5");
+  let next10 = <HTMLInputElement> document.getElementById("next10");
+
+  let from = myTable.from;
+  let totalRecords = getTotalRecords();
+
+  // disable previous buttons when out of range
+  if(from === 0)
+  {
+    previous.disabled=true;
+    previous5.disabled = true;
+    previous10.disabled = true;
+  }
+  else
+  {
+    previous.disabled=false;
+    previous5.disabled=false;
+    previous10.disabled=false;
+  }
+
+  // disable next buttons when out of range
+  if(from + myTable.getNumberOfRows() === (totalRecords -1))
+  {
+    next.disabled=true;
+    next5.disabled=true;
+    next10.disabled=true;
+  }
+  else
+  {
+    next.disabled=false;
+    next5.disabled=false;
+    next10.disabled=false;
+  }
+}
+
+//function to retrieve the total record count used when building the table
+function getTotalRecords()
+{
+    const HttpRequest = new XMLHttpRequest();
+    const url='http://localhost:2050//recordCount';
```

Comment:
Does the string need to be escaped or can this just be
```suggestion
    const url='http://localhost:2050/recordCount';
```